gneiss/regression/_mixedlm.py

```diff
@@ -0,0 +1,270 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, gneiss development team.
+#
+# Distributed under the terms of the GPLv3 License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+import pandas as pd
+import statsmodels.formula.api as smf
+from ._model import RegressionModel
+from ._regression import (_intersect_of_table_metadata_tree,
+                          _to_balances)
+from decimal import Decimal
+from statsmodels.iolib.summary2 import Summary
+from collections import OrderedDict
+
+
+def mixedlm(formula, table, metadata, tree, groups, **kwargs):
+    """ Linear Mixed Effects Models applied to balances.
+
+    A linear mixed effects model is performed on nonzero relative abundance
+    data given a list of covariates, or explanatory variables such as pH,
+    treatment, etc to test for specific effects. The relative abundance data
+    is transformed into balances using the ILR transformation, using a tree to
+    specify the groupings of the features. The linear mixed effects model is
+    applied to each balance separately. Only positive data will be accepted,
+    so if there are zeros present, consider using a zero imputation method
+    such as ``skbio.stats.composition.multiplicative_replacement`` or
+    add a pseudocount.
+
+    Parameters
+    ----------
+    formula : str
+        Formula representing the statistical equation to be evaluated.
+        These strings are similar to how equations are handled in R.
+        Note that the dependent variable in this string should not be
+        specified, since this method will be run on each of the individual
+        balances. See `patsy` for more details.
```

Comment:
I did link to the OLS / mixedlm docs in statsmodels.  I can add an additional reference to patsy
https://patsy.readthedocs.io/en/latest/  

You that should cover it?